src/TwigComponent/src/ComponentPropertiesExtractor.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Twig\PropsNode;
+use Twig\Environment;
+
+final class ComponentPropertiesExtractor
+{
+    public function __construct(
+        private readonly Environment $twig,
+    ) {
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public function getComponentProperties(ComponentMetadata $medata)
+    {
+        if ($medata->isAnonymous()) {
+            return $this->getAnonymousComponentProperties($medata);
+        }
+
+        return $this->getNonAnonymousComponentProperties($medata);
+    }
+
+    /**
+     * @return array<string, string>
+     */
```

Comment:
Not the good type (you return an array of string => array{name: display: type: ...} for no)